docs/why.md

```diff
@@ -0,0 +1,15 @@
+# Why a monorepo?
```

Comment:
Thought: `why.md` isn't a very descriptive name and this focuses on a specific aspect of the architecture. Though it does seem relevant and useful as such.

Here's one thought: Merge `what.md`, `why.md`, and the *Structure* section of `how.md` into `architecture.md` , and this could be a `## Motivation` section there?

Reply:
That would make sense to me. The historical context for how things used to work, and the specific problems we have, will rapidly become less relevant. Explaining the ongoing benefits is useful though.

Reply:
Reminds me of the old README this package had, though that was far longer and even less relevant. There was an elaborate justification given for why we should use TypeScript over JavaScript. I'm sure people needed convincing at the time, but it felt a bit silly after we became more invested in using TS as a team.

Reply:
Good points! I'll try condensing the content and removing the historical angle.